### PR TITLE
Spring 05

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public void removeMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto,
+                                   @PathVariable Long id) {
+        MovieSession movieSession = movieSessionMapper.makeEntity(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.remove(movieSession);
+    }
```
Comment (mfedenko09) [inline src/main/java/com/cinema/controller/MovieSessionController.java]:
```suggestion
    public void removeMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto,
                                   @PathVariable Long id) {
        MovieSession movieSession = movieSessionMapper.makeEntity(movieSessionRequestDto);
        movieSession.setId(id);
        movieSessionService.remove(movieSession);
    }
```
I think, you should delete by id but I not sure

**Review by mfedenko09 — APPROVED** [review]

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/cinema/controller/MovieSessionController.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/cinema/controller/MovieSessionController.java]: done

File: src/main/java/com/cinema/dao/impl/AbstractDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            }
+            throw new DataProcessingException("Unable to add " + entity + "to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    public SessionFactory getSessionFactory() {
```
Comment (boroda4436) [inline src/main/java/com/cinema/dao/impl/AbstractDaoImpl.java]: do we need a getter here? What about a `protected` access modifier?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/dao/impl/AbstractDaoImpl.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/cinema/dao/impl/AbstractDaoImpl.java]: done

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**